demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1051 @@
+                                                                           strlen( sampleazureiotCOMMAND_FAKE_END_TIME ) ) )
+             != eAzureIoTSuccess )
+    {
+        LogError( ( "Error appending end time: result 0x%08x", xResult ) );
+    }
+    else if( ( xResult = AzureIoTJSONWriter_AppendEndObject( pxWriter ) )
+             != eAzureIoTSuccess )
+    {
+        LogError( ( "Error appending end object: result 0x%08x", xResult ) );
+    }
+
+    return xResult;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Command message callback handler
+ */
+static void prvHandleCommand( AzureIoTHubClientCommandRequest_t * pxMessage,
+                              void * pvContext )
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONReader_t xReader;
+    AzureIoTJSONWriter_t xWriter;
+    int32_t lCommandNameLength;
+    int32_t ulCommandPayloadLength;
+    AzureIoTHubClient_t * pxHandle = ( AzureIoTHubClient_t * ) pvContext;
+
+    LogInfo( ( "Command payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               pxMessage->pvMessagePayload ) );
+
+    lCommandNameLength = strlen( sampleazureiotCOMMAND_MAX_MIN_REPORT );
+
+    if( ( lCommandNameLength == pxMessage->usCommandNameLength ) &&
+        ( strncmp( sampleazureiotCOMMAND_MAX_MIN_REPORT, ( const char * ) pxMessage->pucCommandName, lCommandNameLength ) == 0 ) )
+    {
+        /* Is for max min report */
+        xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        xResult = AzureIoTJSONWriter_Init( &xWriter, ucCommandPayloadBuffer, sizeof( ucCommandPayloadBuffer ) );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        xResult = prvInvokeMaxMinCommand( &xReader, &xWriter );
+
+        if( xResult == eAzureIoTSuccess )
+        {
+            ulCommandPayloadLength = AzureIoTJSONWriter_GetBytesUsed( &xWriter );
+
+            if( ulCommandPayloadLength > 0 )
+            {
+                if( AzureIoTHubClient_SendCommandResponse( pxHandle, pxMessage, 200,
+                                                           ucCommandPayloadBuffer,
+                                                           ulCommandPayloadLength ) != eAzureIoTSuccess )
+                {
+                    LogError( ( "Error sending command response" ) );
+                }
+            }
+        }
+        else
+        {
+            LogError( ( "Error generating command payload: 0x%08x", xResult ) );
+
+            if( AzureIoTHubClient_SendCommandResponse( pxHandle, pxMessage, 501,
+                                                       ( const uint8_t * ) sampleazureiotCOMMAND_EMPTY_PAYLOAD,
+                                                       strlen( sampleazureiotCOMMAND_EMPTY_PAYLOAD ) ) != eAzureIoTSuccess )
+            {
+                LogError( ( "Error sending command response" ) );
```

Comment:
On LogErrors generally, we should be capturing the return code from our function and outputting it.  We're pretty thoughtful on these errors so showing this pattern would help log based debugging.

Reply:
added here and elsewhere in the file